### ISSUE TITLE
ADD does not extract zip archives, so DL3020 should not allow them.

### ISSUE THREAD

**Issue by kevinmehall** [hadolint/hadolint#676]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

The [DL3020](https://github.com/hadolint/hadolint/wiki/DL3020) (Use COPY instead of ADD for files and folders) rule is intended to allow `ADD` only for the purposes of extracting archives, recommending `COPY` when not doing so. It appears to explicitly allow filenames ending in `.zip`; however `ADD` [only supports tar archives](https://docs.docker.com/engine/reference/builder/#add) and does not support zip:

> If <src> is a local tar archive in a recognized compression format (identity, gzip, bzip2 or xz) then it is unpacked as a directory. 

Technically, it's the contents of the file that matter, not the extension, but the extension is a reasonable guess if you want to look at the Dockerfile only.

### Expected behavior

```
ADD foo.zip /foo/
```

should yield a DL3020 warning because `ADD` only supports tar archives, not zip, so the zip file will be copied into the image as-is, and `COPY` would work just as well.

### Actual behavior

No warning given when the extension is .zip
